crates/bevy_asset/src/id.rs

```diff
@@ -26,15 +26,20 @@ pub enum AssetId<A: Asset> {
     ///
     /// [`Assets`]: crate::Assets
     Index {
+        /// The unstable, opaque index of the asset.
```

Comment:
This comment sent me down a giant rabbit whole lol. 

I get that opaque means that there is some data known to the compiler but not to the user. I don't really know what unstable means in this context though. 

Maybe it's worth linking to the AssetIndex struct for users to read more?

Reply:
Opaque means you can't "see inside" it, or e.g. do arithmetic with it, but only use it by giving it to APIs. The compiler helps us enforce this but that's not a requirement for using the term.

Unstable means you can't expect to get the same index between restarts of the application, so it doesn't make sense to serialize it for example.

In all cases I can think of where doc links are rendered you'd also have the type right there, so a link is probably redundant.